sippy/UaStateConnected.py

```diff
@@ -151,13 +151,13 @@ def recvEvent(self, event):
         if isinstance(event, CCEventDisconnect) or isinstance(event, CCEventFail) or isinstance(event, CCEventRedirect):
             #print 'event', event, 'received in the Connected state sending BYE'
             if not isinstance(event, CCEventFail):
-                redirect = event.getData()
+                redirects = event.getData()
             else:
-                redirect = None
-            if redirect != None and self.ua.useRefer:
+                redirects = None
+            if redirects != None and self.ua.useRefer:
                 req = self.ua.genRequest('REFER', reason = event.reason)
                 self.ua.lCSeq += 1
-                also = SipReferTo(address = SipAddress(url = redirect))
+                also = SipReferTo(redirects[0])
```

Comment:
This is incorrect. When multiple redirects are provided, one with a highest value needs to be selected, not the first one. For that reason I suggested a method to be implemented in the CCEventXX to retrieve redirects in the proper order. Alternatively, sorting can be done during the generation of the event to avoid hitting parsing issues when originating outbound call.

Reply:
That all being said, there is no way we can logically get more than one redirect in the CCEventDisconnect (see above), so that the whole code can be removed.